test/units/testsuite-71.sh

```diff
@@ -89,10 +89,25 @@ test_chassis() {
     fi
 }
 
+
+restore_sysfs_dmi() {
+    umount /sys/class/dmi/id
+}
+
+
+test_firmware_date() {
+    trap restore_machine_info RETURN
+
+    mount -t tmpfs none /sys/class/dmi/id
+    echo '12/31/2022' > /sys/class/dmi/id/bios_date
+    assert_in '2022-12-31' "$(hostnamectl)"
```

Comment:
hostnamed might cache the info. hence issue `systemctl restart systemd-hostnamed` before, so that the cache is flushed and the date re-read again